jenkins/azure-pipelines.yml

```diff
@@ -76,16 +104,33 @@ jobs:
   - task: PublishBuildArtifacts@1
     displayName: 'Upload Packages'
     inputs:
-      PathtoPublish: '../package'
+      PathtoPublish: './package'
       ArtifactName: 'drop'
       publishLocation: 'Container'
 
   - script: |
       set -e
       set -x
-      ./jenkins/run-tests.sh
+      ./binding-tools-for-swift/jenkins/run-tests.sh
     displayName: 'Run tests'
 
+  - task: PublishTestResults@2
+    displayName: 'Publish NUnit Device Test Results'
+    inputs:
+      testResultsFormat: NUnit
+      testResultsFiles: '**/TestResult*.xml'
+      failTaskOnFailedTests: true
+    continueOnError: true
+    condition: succeededOrFailed()
+
+  - task: PublishPipelineArtifact@1
+    displayName: 'Publish Artifact: TestResult'
+    inputs:
+      targetPath: $(System.DefaultWorkingDirectory)/binding-tools-for-swift/tests/TestResults.md
+      artifactName: TestResults
+    continueOnError: true
+    condition: succeededOrFailed()
```

Comment:
This looks great! The only thing I'd be extra careful with is checking that all of your conditions run as expected. From the docs, `Only when all previous dependencies have succeeded. This is the default if there is not a condition set in the YAML.` For example, if "upload packages" fails would you still want to run the tests? (as-is, I *think* that you'd skip to "publish unit device test results" since that runs regardless of previous success/failure.

I ran into some issues when I was in YAML b/c I didn't explicitly specify the condition for some steps so tasks stopped running earlier in the process than we wanted :(

https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml

Reply:
Good catch! I'm not too sure at the moment which dependencies the tests need. I am assuming uploading the packages are not too important to the tests but I am guessing the steps previous to that are important for tests. I will look further into that. Thanks!